app/Chatkit Quickstart/MessagesDataModel.swift

```diff
@@ -18,16 +21,45 @@ struct LocalMessage {
     let text: String
     let internalId: String
     
+    init(text: String) {
+        self.text = text
+        self.internalId = UUID().uuidString
+    }
+    
     var parts: [PCPartRequest] {
-        get {
-            return [
-                PCPartRequest(.inline(PCPartInlineRequest(type: MIME_TYPE_TEXT, content: text))),
-                PCPartRequest(.inline(PCPartInlineRequest(type: MIME_TYPE_INTERNAL_ID, content: internalId)))            
-            ]
+        return [
+            PCPartRequest(.inline(PCPartInlineRequest(type: MIME_TYPE_TEXT, content: text))),
+            PCPartRequest(.inline(PCPartInlineRequest(type: MIME_TYPE_INTERNAL_ID, content: internalId)))
+        ]
+    }
+}
+
+extension PCMultipartMessage {
+    
+    var internalId: String? {
+        return part(forType: MIME_TYPE_INTERNAL_ID)
+    }
+    
+    var text: String {
+        return part(forType: MIME_TYPE_TEXT) ?? ""
+    }
+    
+    private func part(forType type: String) -> String? {
+        for part in parts {
+            if case .inline(let payload) = part.payload {
+                if payload.type == type {
+                    return payload.content
+                }
+            }
         }
+        return nil
     }
 }
 
+protocol MessagesDataModelDelegate {
+    func didChange(model: MessagesDataModel.MessagesModel, changeType: ChangeType)
```

Comment:
```suggestion
    func messagesDataModel(_ messagesDataModel: MessagesDataModel, didAddMessagesModel messagesModel: MessagesDataModel.MessagesModel, atIndex index: Int)
    func messagesDataModel(_ messagesDataModel: MessagesDataModel, didUpdateMessagesModel messagesModel: MessagesDataModel.MessagesModel, atIndex index: Int)
```

Reply:
Good suggestion.  I have committed this change as discussed.